gen/api_defs.jl

```diff
@@ -18,248 +18,248 @@
 
 ###
 ### Error Interface
 ###
 
-@bind h5e_get_auto(estack_id::Hid, func::Ref{Ptr{Cvoid}}, client_data::Ref{Ptr{Cvoid}})::Herr "Error getting error reporting behavior"
-@bind h5e_set_auto(estack_id::Hid, func::Ptr{Cvoid}, client_data::Ptr{Cvoid})::Herr "Error setting error reporting behavior"
+@bind h5e_get_auto(estack_id::hid_t, func::Ref{Ptr{Cvoid}}, client_data::Ref{Ptr{Cvoid}})::herr_t "Error getting error reporting behavior"
+@bind h5e_set_auto(estack_id::hid_t, func::Ptr{Cvoid}, client_data::Ptr{Cvoid})::herr_t "Error setting error reporting behavior"
 
 ###
 ### File Interface
 ###
 
-@bind h5f_close(file_id::Hid)::Herr "Error closing file"
-@bind h5f_create(pathname::Ptr{UInt8}, flags::Cuint, fcpl_id::Hid, fapl_id::Hid)::Hid error("Error creating file ", pathname)
-@bind h5f_flush(object_id::Hid, scope::Cint)::Herr "Error flushing object to file"
-@bind h5f_get_access_plist(file_id::Hid)::Hid "Error getting file access property list"
-@bind h5f_get_create_plist(file_id::Hid)::Hid "Error getting file create property list"
-@bind h5f_get_intent(file_id::Hid, intent::Ptr{Cuint})::Herr "Error getting file intent"
-@bind h5f_get_name(obj_id::Hid, buf::Ptr{UInt8}, buf_size::Csize_t)::Cssize_t "Error getting file name"
-@bind h5f_get_obj_count(file_id::Hid, types::Cuint)::Cssize_t "Error getting object count"
-@bind h5f_get_obj_ids(file_id::Hid, types::Cuint, max_objs::Csize_t, obj_id_list::Ptr{Hid})::Cssize_t "Error getting objects"
-@bind h5f_get_vfd_handle(file_id::Hid, fapl_id::Hid, file_handle::Ptr{Ptr{Cint}})::Herr "Error getting VFD handle"
-@bind h5f_is_hdf5(pathname::Cstring)::Htri error("Cannot access file ", pathname)
-@bind h5f_open(pathname::Cstring, flags::Cuint, fapl_id::Hid)::Hid error("Error opening file ", pathname)
-@bind h5f_start_swmr_write(id::Hid)::Herr "Error starting SWMR write"
+@bind h5f_close(file_id::hid_t)::herr_t "Error closing file"
+@bind h5f_create(pathname::Ptr{UInt8}, flags::Cuint, fcpl_id::hid_t, fapl_id::hid_t)::hid_t error("Error creating file ", pathname)
+@bind h5f_flush(object_id::hid_t, scope::Cint)::herr_t "Error flushing object to file"
+@bind h5f_get_access_plist(file_id::hid_t)::hid_t "Error getting file access property list"
+@bind h5f_get_create_plist(file_id::hid_t)::hid_t "Error getting file create property list"
+@bind h5f_get_intent(file_id::hid_t, intent::Ptr{Cuint})::herr_t "Error getting file intent"
+@bind h5f_get_name(obj_id::hid_t, buf::Ptr{UInt8}, buf_size::Csize_t)::Cssize_t "Error getting file name"
+@bind h5f_get_obj_count(file_id::hid_t, types::Cuint)::Cssize_t "Error getting object count"
+@bind h5f_get_obj_ids(file_id::hid_t, types::Cuint, max_objs::Csize_t, obj_id_list::Ptr{hid_t})::Cssize_t "Error getting objects"
+@bind h5f_get_vfd_handle(file_id::hid_t, fapl_id::hid_t, file_handle::Ptr{Ptr{Cint}})::herr_t "Error getting VFD handle"
+@bind h5f_is_hdf5(pathname::Cstring)::htri_i error("Cannot access file ", pathname)
```

Comment:
```suggestion
@bind h5f_is_hdf5(pathname::Cstring)::htri_t error("Cannot access file ", pathname)
```
Looks like typo here and in a couple of other places.